Mage/src/main/java/mage/abilities/Ability.java

```diff
@@ -157,6 +158,14 @@ default List<String> getManaCostSymbols() {
 
     void addManaCostsToPay(ManaCost manaCost);
 
+    /**
+     * Gets a map of the cost tags (set while casting/activating)
+     * Note that tag entries should always be immutable
+     *
+     * @return The map of tags and corresponding objects (usually Integer or Boolean, sometimes other)
+     */
+    Map<String, Object> getCostsTagMap();
```

Comment:
What types of data it will be contains in the future?

Reply:
Convoke stores a HashSet of the creatures used to convoke it.
> 707.10 [...] If an effect of the copy refers to objects used to pay its costs, it uses the objects used to pay the costs of the original spell or ability.

I also used some null keys where the condition was the presence of the tag at all.

Potentially in the future it could also be used for things like storing creatures sacrificed to pay a spell's cost (with the goal of eventually removing all need for looping through a spell's costs), but that is not going to be part of this set of PRs.